tests/code/test_events.py

```diff
@@ -519,33 +522,92 @@ def test_getAllFacilitators():
+                                description = "Test upcoming no program event.",
+                                timeStart = "18:00:00",
+                                timeEnd = "21:00:00",
+                                location = "The moon",
+                                startDate = "2021-12-12",
+                                endDate = "2021-12-13")
+
+        # Create a Program Event to show up when the user marks interest in a
+        # new program
+        newProgramEvent = Event.create(name = "Upcoming event with  program",
+                                term = 2,
+                                description = "Test upcoming program event.",
+                                timeStart = "18:00:00",
+                                timeEnd = "21:00:00",
+                                location = "The sun",
+                                startDate = 2021-12-12,
+                                endDate = 2021-12-13)
+
+        # Create a new Program to create the new Program Event off of so the
+        # user can mark interest for it
+        ProgramForInterest = Program.create(id = 13,
+                                            programName = "BOO",
+                                            isStudentLed = False,
+                                            isBonnerScholars = False,
+                                            emailReplyTo = "test@email",
+                                            emailSenderName = "testName")
+
+        ProgramEventForInterest = ProgramEvent.create(program = 13, event = newProgramEvent)
+
+        program_id = 13
+
+        # User has not RSVPd and is Interested
+        addInterest = addUserInterest(program_id, user)
+        eventsInUserInterestedProgram = getUpcomingEventsForUser(user, asOf = testDate)
+
+        eventsInProgram = list((Event.select()
+                                     .join(ProgramEvent)
+                                     .join(Program)
+                                     .where(Event.startDate >= testDate,
+                                            ProgramEvent.event_id == Event.id,
+                                            ProgramEvent.program_id == 13)))
+
+        assert eventsInUserInterestedProgram == eventsInProgram
+
+        # user has RSVPd and is Interested
+        addUserRsvp = addVolunteerToEventRsvp(user, noProgram.id)
+        eventsInUserInterestAndRsvp = getUpcomingEventsForUser(user, asOf = testDate)
+
+        eventsInProgramAndRsvp = list((Event.select()
+                                        .join(ProgramEvent, JOIN.LEFT_OUTER)
+                                        .join(Interest, JOIN.LEFT_OUTER, on=(ProgramEvent.program == Interest.program))
+                                        .join(EventRsvp, JOIN.LEFT_OUTER, on=(Event.id == EventRsvp.event))
+                                        .where(Event.startDate >= testDate,
+                                              (Interest.user == user) | (EventRsvp.user == user))))
```

Comment:
Same comment here. You don't need to duplicate the query because you know what will be included